pkg/util/client.go

```diff
@@ -10,17 +11,27 @@ import (
 )
 
 func LoadClusterConfig() (*rest.Config, error) {
+	if env := os.Getenv(clientcmd.RecommendedConfigPathEnvVar); env != "" {
+		// someone is explicitly passing a KUBECONFIG, use it
+		return loadClusterConfig()
+	}
+
+	// otherwise, prefer in-cluster config, then fall back to normal loading
 	clusterConfig, err := rest.InClusterConfig()
 	if err == nil {
 		return clusterConfig, nil
 	}
 
+	return loadClusterConfig()
```

Comment:
What? If env var is set use that, if not use inCluster, do not show failure, and go back to step 1? This doesn't make sense
/hold

Reply:
Something like this, with an added pre-condition that sets the kubeconfigs flag value to the env var if unset: https://github.com/kubermatic/machine-controller/blob/23656f1f2c518c1e555b12362501b835d36794c3/cmd/machine-controller/main.go#L210

Or just the c-r stuff which does all of it for you.